src/api/lib/api_error.rb

```diff
@@ -38,7 +38,32 @@ def default_message
   end
 end
 
-# 403 errors (how about a subclass?)
+# 401 errors
+class AuthenticationFailed < APIError
+  setup 401, 'Authentication Failed'
+end
+
+class AuthenticationRequiredError < APIError
+  setup 401, 'Authentication Required'
+end
+
+class InactiveUserError < APIError
+  setup 403, 'Your account is in a not active state. Talk to your OBS Admin.'
+end
+
+class UnconfirmedUserError < APIError
+  setup 403, 'Your account is not yet approved. Talk to your OBS Admin.'
+end
+
+class AnonymousUser < APIError
+  setup 401, 'Anonymous user is not allowed here - please login'
+end
+
```

Comment:
```suggestion
```

After #18266, I guess this exception should be removed or replaced.

I've just noticed that there is a leftover in master: https://github.com/openSUSE/open-build-service/blob/96ef9e1e06003b2fb872c48c5f9145a806ed6d5e/src/api/app/controllers/public_controller.rb#L233

Reply:
true. dropped again!